metrics-exporter/src/bin/io_engine/client/grpc_client.rs

```diff
@@ -75,71 +66,22 @@ impl GrpcClient {
     pub(crate) async fn new(context: GrpcContext) -> Result<Self, ExporterError> {
         let sleep_duration_sec = 10;
         loop {
-            match context.api_version {
-                ApiVersion::V0 => {
-                    match tokio::time::timeout(
-                        context.timeouts.connect(),
-                        MayaClientV0::connect(context.endpoint.clone()),
-                    )
-                    .await
-                    {
-                        Err(error) => {
-                            error!(error=%error, "Grpc connection timeout, retrying after {}s",sleep_duration_sec);
-                        }
-                        Ok(result) => match result {
-                            Ok(v0_client) => {
-                                return Ok(Self {
-                                    ctx: context.clone(),
-                                    v0_client: Some(v0_client),
-                                    v1_client: None,
-                                })
-                            }
-                            Err(error) => {
-                                error!(error=%error, "Grpc client connection error, retrying after {}s",sleep_duration_sec);
-                            }
-                        },
-                    }
-                }
-                ApiVersion::V1 => {
-                    match tokio::time::timeout(
-                        context.timeouts.connect(),
-                        PoolClient::connect(context.endpoint.clone()),
-                    )
-                    .await
-                    {
-                        Err(error) => {
-                            error!(error=%error, "Grpc connection timeout, retrying after {}s",sleep_duration_sec);
-                        }
-                        Ok(result) => match result {
-                            Ok(pool) => {
-                                info!("grpc connected successfully");
-                                return Ok(Self {
-                                    ctx: context.clone(),
-                                    v0_client: None,
-                                    v1_client: Some(MayaClientV1 { pool }),
-                                });
-                            }
-                            Err(error) => {
-                                error!(error=%error, "Grpc client connection error, retrying after {}s",sleep_duration_sec);
-                            }
-                        },
-                    }
-                }
+            if let Ok(channel) = context.endpoint.connect().await {
+                let pool = PoolClient::new(channel.clone());
+                let stats = StatsClient::new(channel.clone());
+                return Ok(Self {
+                    v1_client: Some(MayaClientV1 { pool, stats }),
+                });
+            } else {
+                error!(
```

Comment:
Shall we make this  silenced until we've retried a number of times?
This confuses a lot of users because this container can start before the io-engine.